tests/functional/notification/NotificationConfigManager.js

```diff
@@ -0,0 +1,161 @@
+const assert = require('assert');
+const async = require('async');
+const werelogs = require('werelogs');
+const ZookeeperMock = require('zookeeper-mock');
+
+const NotificationConfigManager
+    = require('../../../extensions/notification/NotificationConfigManager');
+
+const logger = new werelogs.Logger('NotificationConfigManager:test');
+const zkConfigParentNode = 'config';
+const concurrency = 10;
+const bucketPrefix = 'bucket';
+const timeoutMs = 100;
+
+function getTestConfigValue(bucket) {
+    const data = {
+        bucket,
+        notificationConfiguration: {
+            queueConfig: [
+                {
+                    events: ['s3:ObjectCreated:Put'],
+                    queueArn: 'arn:scality:bucketnotif:::destination1',
+                    filterRules: [],
+                    id: `${zkConfigParentNode}-${bucket}`,
+                },
+            ],
+        },
+    };
+    return data;
+}
+
+function populateTestConfigs(zkClient, numberOfConfigs, cb) {
+    async.timesLimit(numberOfConfigs, concurrency, (n, done) => {
+        const bucket = `${bucketPrefix}${n + 1}`;
+        const val = getTestConfigValue(bucket);
+        const strVal = JSON.stringify(val);
+        const node = `/${zkConfigParentNode}/${bucket}`;
+        async.series([
+            nxt => zkClient.mkdirp(node, nxt),
+            // nxt => zkClient.setData(node, strVal, nxt),
+            nxt => zkClient.setData(node, Buffer.from(strVal), nxt),
+        ], done);
+    }, cb);
+}
+
+function listBuckets(zkClient, cb) {
+    const node = `/${zkConfigParentNode}`;
+    zkClient.getChildren(node, cb);
+}
+
+function deleteTestConfigs(zkClient, cb) {
+    const node = `/${zkConfigParentNode}`;
+    listBuckets(zkClient, (e, children) => {
```

Comment:
Same idea, why `e` not `err` or `error` which is more descriptive while being reasonably short, and compels with 99% of our usage throughout the code